### PR TITLE
Improved type definitions for Recharts

### PR DESCRIPTION
**Pull request by mthmulders**
- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: Correct import style shown [here](http://recharts.org/#/en-US/guide/getting-started)
- [X] Increase the version number in the header if appropriate.
- [X] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by rapmue** [comment]
I think @dt-bot has missed my PR to your branch @mthmulders . Do we have to do something about this?

File: types/recharts/tslint.json [hunk 1]
```diff
@@ -1,3 +1,7 @@
 {
-  "extends": "dtslint/dt.json"
+	"extends": "dtslint/dt.json",
+	"rules": {
+		"no-single-declare-module": false,
```
Comment (weswigham) [inline types/recharts/tslint.json]: Remove this rule ignore.

File: types/recharts/tslint.json [hunk 2]
```diff
@@ -1,3 +1,7 @@
...
+		"strict-export-declare-modifiers": false
```
Comment (weswigham) [inline types/recharts/tslint.json]: And this one.

File: types/recharts/index.d.ts [hunk 3]
```diff
@@ -1,25 +1,38 @@
 // Type definitions for Recharts 0.22
 // Project: http://recharts.org/
 // Definitions by: Maarten Mulders <https://github.com/mthmulders/>
+//                 Raphael Mueller <https://github.com/rapmue/>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
-import * as React from 'react';
+declare module 'recharts' {
```
Comment (weswigham) [inline types/recharts/index.d.ts]: Then just remove this module wrapper from the file. Nothing else should need to be changed.

**Review by weswigham — CHANGES_REQUESTED** [review]

File: types/recharts/index.d.ts (hunk 3 above)
Comment (mthmulders) [inline types/recharts/index.d.ts]: @rapmue I'm not completely sure about this one. What's your thought?

**Comment by mthmulders** [comment]
@rapmue I think that @dt-bot only monitors PR's to this repo, not to mine. That's why I merged your PR into my repo, so that now we can merge my repo into this one. I think all of your changes are in this PR, but correct me if I'm wrong.

**Comment by mthmulders** [comment]
@weswigham, as you can see I can remove the two lines from tslint.json, but it breaks the Travis build. To use Recharts correctly, the [docs](http://recharts.org/#/en-US/guide/getting-started) say you should import its types using `import { LineChart, Line } from 'recharts'`. How to get that working using the default tslint settings?

**Comment by weswigham** [comment]
@mthmulders just remove the `declare module 'recharts'` and accompanying curly braces wrapping all your declarations. The file itself acts as your module declaration.

**Comment by mthmulders** [comment]
@weswigham Thanks for the hint, that seems to fix it. Need to export the interface props as well, although that is not strictly required for Recharts but otherwise tslint will say
> Prefer to explicitly write 'export' for an external module export.

File: types/recharts/index.d.ts (hunk 3 above)
Comment (mthmulders) [inline types/recharts/index.d.ts]: Nevermind, got it sorted already.

**Comment by weswigham** [comment]
Great, these look excellent now

**Comment by mthmulders** [comment]
Thanks for the help, @weswigham!

**Pull request merged**